osu.Game/Utils/ModUtils.cs

```diff
@@ -106,20 +106,40 @@ public static bool CheckAllowed(IEnumerable<Mod> combination, IEnumerable<Type>
         }
 
         /// <summary>
-        /// Check the provided combination of mods are valid for a local gameplay session.
+        /// Checks that all <see cref="Mod"/>s in a combination are valid for a local gameplay session.
         /// </summary>
         /// <param name="mods">The mods to check.</param>
-        /// <param name="invalidMods">Invalid mods, if any were found. Can be null if all mods were valid.</param>
+        /// <param name="invalidMods">Invalid mods, if any were found. Will be null if all mods were valid.</param>
         /// <returns>Whether the input mods were all valid. If false, <paramref name="invalidMods"/> will contain all invalid entries.</returns>
         public static bool CheckValidForGameplay(IEnumerable<Mod> mods, [NotNullWhen(false)] out List<Mod>? invalidMods)
+            => checkValid(mods, m => m.Type != ModType.System && m.HasImplementation && !(m is MultiMod), out invalidMods);
```

Comment:
Is there a reason why this isn't going through `IsPlayable(User)`?

Reply:
It wasn't checked for previously, but can add it there.